clippy_utils/src/macros.rs

```diff
@@ -461,11 +462,11 @@ impl<'tcx> FormatArgsExpn<'tcx> {
             .collect()
     }
 
-    /// Span of all inputs
+    /// Source callsite span of all inputs
     pub fn inputs_span(&self) -> Span {
         match *self.value_args {
             [] => self.format_string_span,
-            [.., last] => self.format_string_span.to(last.span),
+            [.., last] => self.format_string_span.to(last.span.source_callsite()),
```

Comment:
If the `format_args!(..)` call is inside a macro definition, then we don't want to walk all the way to the source root.
```suggestion
            [.., last] => self.format_string_span.to(hygiene::walk_chain(last.span, self.format_string_span.ctxt())),
```

Reply:
👍

Changed